docs: provide additional setup instructions

josephmcasey opened the pull request:
Resolves https://github.com/PrefectHQ/prefect-operator/issues/148

Commit:
docs: provide additional setup instructions

Resolves https://github.com/PrefectHQ/prefect-operator/issues/148

mitchnielsen reviewed (changes requested):
Thanks for the contribution here @josephmcasey, just a couple notes 👍🏼

josephmcasey commented on README.md:
````diff
@@ -14,6 +14,9 @@ It follows the [Kubernetes operator pattern](https://kubernetes.io/docs/concepts
 Dependencies can be installed with [`mise`](https://mise.jdx.dev):
 
 ```shell
+mise settings experimental=true
````
This is a much better solution to this issue, since it reduces the number of commands a user has to execute themselves. I've incorporated your suggests as you have written it.

josephmcasey commented on README.md:
````diff
@@ -14,6 +14,9 @@ It follows the [Kubernetes operator pattern](https://kubernetes.io/docs/concepts
 
 ```shell
+mise settings experimental=true
+mise use -g python
+pip install --user pipx
````
I wrote this note as part of the prerequisites for the project. Can you perform an editorial review and let me know if the updated documentation is satisfactory?

mitchnielsen replied:
```suggestion
```

Let's remove `mise use -g python` and instead put the Python dependency in `.mise.toml` directly:

```diff
diff --git a/.mise.toml b/.mise.toml
index cb2aa8b..7aeda38 100644
--- a/.mise.toml
+++ b/.mise.toml
@@ -15,6 +15,7 @@ setup-envtest = '0.17.0'
 shellcheck = '0.10.0'
 yamllint = '1.35.1'
 yq = '4.44.3'
+python = "3.11"
```

I also removed the `pipx` installation - I don't believe that's required from my local testing and I don't see it mentioned in https://mise.jdx.dev/lang/python.html#usage. Do you agree?

josephmcasey replied:
From a clean MacBook, these are screenshots evidentiating the commands required to run from brand new state.

```sh
curl https://mise.run | sh
mise trust
pip install --user pipx
mise install
```

### Screenshots
<img width="639" alt="clean-state-install-1" src="https://github.com/user-attachments/assets/bf33ffb4-dcf8-48dd-bdd7-76eca23a6a42" />
<img width="638" alt="clean-state-install-2" src="https://github.com/user-attachments/assets/3e16270d-e281-4f77-85ed-47dd77d27a2a" />
<img width="637" alt="clean-state-install-3" src="https://github.com/user-attachments/assets/e9f61b1a-b469-426d-b8d2-bc7becf83bac" />

mitchnielsen replied:
I'm a bit surprised at the secondary dependencies they used for their installer. Thanks for confirming.

Commit:
docs: update README.md
- Provide documentation explaining the necessity of certain pythonic tooling libraries to ensure proper functionality of the mise tooling used for overall environmental and project management for development purposes.

Commit:
ci: configure mise to use experimental features
- Current usage of the mise tooling requires users to install go dependencies. This requires experimental configurations to be enabled to ensure their proper installation

mitchnielsen commented on README.md:
````diff
@@ -10,10 +10,15 @@ It follows the [Kubernetes operator pattern](https://kubernetes.io/docs/concepts
 Dependencies can be installed with [`mise`](https://mise.jdx.dev):
 
 ```shell
+mise settings experimental=true
````
```suggestion
```

Removes this command now that it's specified in `.mise.toml`.

josephmcasey replied:
Good catch. I've removed that test. I found one command that I missed which deserves your attention. It's `mise trust`.

Commit:
docs: rewrite 'Getting Stated' section of README.md
- remove redundant setup commands
- split prerequisite tooling from project setup commadns

mitchnielsen commented:
Thank you @josephmcasey 🤝
